cmd/account/mgmt/account-unassign.go

```diff
@@ -128,57 +132,119 @@ func (o *accountUnassignOptions) run() error {
 		os.Exit(0)
 	}
 
-	// loop through accounts list and untag and move them back into root OU
 	for _, id := range accountIdList {
-
+		// untag account
 		err = o.untagAccount(id)
 		if err != nil {
 			return err
 		}
-
+		// move account
 		err = o.moveAccount(id, rootID, destinationOU)
 		if err != nil {
 			return err
 		}
+		// instantiate new client with AssumeRole
+		newAWSClient, err := o.assumeRoleForAccount(id)
+		if err != nil {
+			return err
+		}
+		// delete roles
+		err = o.deleteRoles(newAWSClient)
+		if err != nil {
+			return err
+		}
+		// list iam users created by each account and append to slice
+		users, err := o.listUsersFromAccount(newAWSClient, id)
+		if err != nil {
+			return err
+		}
+
+		allUsers = append(allUsers, users...)
+
 	}
 
-	// Delete login profile
-	err = o.deleteLoginProfile(accountUsername)
-	if err != nil {
-		return err
+	for _, userName := range allUsers {
+
+		// Delete login profile
+		err = o.deleteLoginProfile(userName)
+		if err != nil {
+			return err
+		}
+		// Delete access keys
+		err = o.deleteAccessKeys(userName)
+		if err != nil {
+			return err
+		}
+		// Delete signing certificates
+		err = o.deleteSigningCert(userName)
+		if err != nil {
+			return err
+		}
+		// Delete policies
+		err = o.deletePolicies(userName)
+		if err != nil {
+			return err
+		}
+		// Delete attached policies
+		err = o.deleteAttachedPolicies(userName)
+		if err != nil {
+			return err
+		}
+		// Delete groups
+		err = o.deleteGroups(userName)
+		if err != nil {
+			return err
+		}
+		// Delete user
+		err = o.deleteUser(userName)
+		if err != nil {
+			return err
+		}
 	}
-	// Delete access keys
-	err = o.deleteAccessKeys(accountUsername)
-	if err != nil {
-		return err
+
+	return nil
+}
+
+func (o *accountUnassignOptions) assumeRoleForAccount(account_id string) (awsprovider.Client, error) {
+
+	roleArn := fmt.Sprintf("arn:aws:%s:iam::role/OrganizationAccountAccessRole", account_id)
+	input := &sts.AssumeRoleInput{
+		RoleArn:         aws.String(roleArn),
+		RoleSessionName: aws.String("osdctl-account-unassignment"),
 	}
-	// Delete signing certificates
-	err = o.deleteSigningCert(accountUsername)
+	result, err := o.awsClient.AssumeRole(input)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	// Delete policies
-	err = o.deletePolicies(accountUsername)
-	if err != nil {
-		return err
+	newAwsClientInput := &awsprovider.AwsClientInput{
+		AccessKeyID:     *result.Credentials.AccessKeyId,
+		SecretAccessKey: *result.Credentials.SecretAccessKey,
+		SessionToken:    *result.Credentials.SessionToken,
+		Region:          "us-east-1",
```

Comment:
Can this be a variable from the default region used? I. Am not sure this will work in all cases (feel free to correct me if I am off base)

Reply:
IAM resources are global, and not locked to a region. The AWS client needs a region though, so us-east-1 is generally "default".

Reply:
resolving as this is is a "aws api issue"